### PR TITLE
Update the lib to 6.5.0 and add `gradle-check-flaky-test-issue-creation.jenkinsfile`.

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description

- Update the lib to 6.5.0 to fix the issue with `gh issue list`, refer https://github.com/opensearch-project/opensearch-build-libraries/issues/441. The PR to fix this is https://github.com/opensearch-project/opensearch-build-libraries/pull/442.

- Add `gradle-check-flaky-test-issue-creation.jenkinsfile` to fix the issue https://github.com/opensearch-project/OpenSearch/issues/13950. This jenkinsfile uses the library `gradleCheckFlakyTestChecker ` that is part of PR https://github.com/opensearch-project/opensearch-build-libraries/pull/436.

### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/13950 and https://github.com/opensearch-project/opensearch-build-libraries/issues/441

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
The 6.5.0 library is just released https://github.com/opensearch-project/opensearch-build-libraries/releases/tag/6.5.0.

File: jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    triggers {
+        parameterizedCron '''
+            H */8 * * *
+        '''
+    }
+    stages {
+        stage('Detect Gradle Check Flaky Tests') {
+            steps {
+                script {
+                    gradleCheckFlakyTestChecker()
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile:31]: With this limitation currently its not possible https://github.com/opensearch-project/opensearch-build-libraries/pull/436#issuecomment-2164019900.

[the remark this answers is not in the document]
Comment (gaiksaya) [reply jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile:31]: So even the regression txt file wont be generated?

[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile:31]: It wont Sayali initially I have tried it, but the line it reaches `gradleCheckFlakyTestChecker` it starts failing for this line here https://github.com/opensearch-project/opensearch-build-libraries/blob/main/src/gradlecheck/FetchPostMergeFailedTestClass.groovy#L77 as it's not able to find the class `OpenSearchMetricsQuery`.

[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile:31]: Tried the following way but fails at `OpenSearchMetricsQuery`, its not able to mock external classes part of the used library.
```
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import jenkins.tests.BuildPipelineTest
import org.junit.Before
import org.junit.Test
import org.yaml.snakeyaml.Yaml
import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
import static com.lesfurets.jenkins.unit.global.lib.GitSource.gitSource
import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
import static org.hamcrest.CoreMatchers.hasItem
import static org.hamcrest.MatcherAssert.assertThat
import static org.junit.jupiter.api.Assertions.assertThrows

class TestGradleCheckFlakyTestIssueCreation extends BuildPipelineTest { 

    // Define the mocked response for the fetchMetrics method
    def yourMockedResponseHere = '''
    {
        "aggregations": {
            "test_class_keyword_agg": {
                "buckets": [
                    {
                        "key": "TestClass1",
                        "doc_count": 10
                    },
                    {
                        "key": "TestClass2",
                        "doc_count": 5
                    }
                ]
            }
        }
    }
    '''

    @Override
    @Before
    void setUp() {
        helper.registerSharedLibrary(
            library().name('jenkins')
                .defaultVersion('main')
                .allowOverride(true)
                .implicit(true)
                .targetPath('vars')
                .retriever(gitSource('https://github.com/prudhvigodithi/opensearch-build-libraries.git'))
                .build()
        )

        super.setUp()
        addParam('METRICS_HOST_ACCOUNT', '1.0.0')
        helper.registerAllowedMethod('withAWS', [Map, Closure], null)

        // Mock the fetchMetrics method of OpenSearchMetricsQuery
        def openSearchMetricsQueryMock = new OpenSearchMetricsQuery('', '', '', '', null)
        openSearchMetricsQueryMock.metaClass.fetchMetrics = { String query ->
            return new groovy.json.JsonSlurper().parseText(yourMockedResponseHere)
        }
    }

    @Test
    void jenkinsFileExecution() {
        super.testPipeline('jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile',
                'tests/jenkins/jenkinsjob-regression-files/promotion/gradle-check-flaky-test-issue-creation.jenkinsfile')
    }
}

```

[the remark this answers is not in the document]
Comment (gaiksaya) [reply jenkins/gradle/gradle-check-flaky-test-issue-creation.jenkinsfile:31]: Weird! @dblock Maybe you have some idea here?

**Review by gaiksaya — APPROVED** [review]

**Comment by gaiksaya** [comment]
Looks like groovy tests are still failing. Can you please fix them before merging?

**Comment by prudhvigodithi** [comment]
> Looks like groovy tests are still failing. Can you please fix them before merging?

Sure Sayali will take care of it.

**Pull request merged**